src/main/java/com/aerospike/mapper/tools/configuration/ClassConfig.java

```diff
@@ -231,6 +231,16 @@ public Builder withKeyField(String fieldName) {
             return this;
         }
         
+        public Builder withKeyFieldAndStoreAsBin(String fieldName, boolean storeAsBin) {
+            if (this.classConfig.getKey() == null) {
+                this.classConfig.setKey(new KeyConfig());
+            }
+            this.validateFieldExists(fieldName);
+            this.classConfig.getKey().setField(fieldName);
+            this.classConfig.getKey().setStoreAsBin(storeAsBin);
+            return this;
```

Comment:
```suggestion
            this.classConfig.getKey().setStoreAsBin(storeAsBin);
            return withKeyField(fieldName);
```

Reply:
Committed all changes and reverted this specifically -> causing NPE